synapseclient/models/protocols/file_protocol.py

```diff
@@ -0,0 +1,274 @@
+            method for changing the name of the downloaded file.
+        - Moving a file to a new parent by setting the `parent_id` attribute of the
+            File object.
+
+        Arguments:
+            parent: The parent folder or project to store the file in. May also be
+                specified in the File object. If both are provided the parent passed
+                into `store` will take precedence.
+            synapse_client: If not passed in or None this will use the last client from
+                the `.login()` method.
+
+        Returns:
+            The file object.
+
+
+        Example: Using this function
+            File with the ID `syn123` at path `path/to/file.txt`:
+
+                file_instance = File(id="syn123", path="path/to/file.txt").store()
+
+            File at the path `path/to/file.txt` and a parent folder with the ID `syn456`:
+
+                file_instance = File(path="path/to/file.txt", parent_id="syn456").store()
+
+            File at the path `path/to/file.txt` and a parent folder with the ID `syn456`:
+
+                file_instance = File(path="path/to/file.txt").store(parent=Folder(id="syn456"))
+
+            Rename a file (Does not update the file on disk or the name of the downloaded file):
+
+                file_instance = File(id="syn123", download_file=False).get()
+                print(file_instance.name)  ## prints, e.g., "my_file.txt"
+                file_instance.change_metadata(name="my_new_name_file.txt")
+
+            Rename a file, and the name of the file as downloaded (Does not update the file on disk):
+
+                file_instance = File(id="syn123", download_file=False).get()
+                print(file_instance.name)  ## prints, e.g., "my_file.txt"
+                file_instance.change_metadata(name="my_new_name_file.txt", download_as="my_new_name_file.txt")
+
+        """
+        return self
+
+    def change_metadata(
+        self,
+        name: Optional[str] = None,
+        download_as: Optional[str] = None,
+        content_type: Optional[str] = None,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "File":
+        """
+        Change File Entity metadata for properties that are immutable after creation
+        through the store method.
+
+        Arguments:
+            name: Specify filename to change the filename of a file.
+            download_as: Specify filename to change the filename of a filehandle.
+            content_type: Specify content type to change the content type of a filehandle.
+            synapse_client: If not passed in or None this will use the last client from
+                the `.login()` method.
+
+        Returns:
+            The file object.
+
+        Example: Using this function
+            Can be used to change the filename, the filename when the file is downloaded, or the file content-type without downloading:
+
+                file_entity = File(id="syn123").get()
```

Comment:
the example is described as no downloading required but `download_file` isn't set to `False`

Reply:
Corrected: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1068/commits/a40bfdbd525b4b35ec462c2974a91f1f508e849f